scripts/hail_batch/eqtl_hail_batch/generate_eqtl_spearman.py

```diff
@@ -185,11 +150,70 @@ def function_that_merges_dataframes(*df_list):
     return merged_df.to_string()
 
 
-merge_job = b.new_python_job(name='merge_scatters')
-result_second = merge_job.call(
-    function_that_merges_dataframes, *spearman_dfs_from_scatter
+# Create click command line to enter dependency files
+@click.command()
+@click.option(
+    '--expression', required=True, help='A sample x gene TSV of expression values'
+)
+@click.option('--genotype', required=True, help='A TSV of genotypes for each sample')
+@click.option(
+    '--geneloc', required=True, help='A TSV of start and end positions for each gene'
+)
+@click.option(
+    '--snploc',
+    required=True,
+    help='A TSV of snp IDs with chromsome and position values for each',
+)
+@click.option(
+    '--covariates', required=True, help='A TSV of covariates to calculate residuals'
 )
-b.write_output(
-    result_second.as_str(), f'gs://{OUTPUT_BUCKET}/kat/correlation_results.csv'
+@click.option(
+    '--output-prefix',
+    required=True,
+    help='A path prefix of where to output files, eg: gs://MyBucket/output-folder/',
 )
-b.run()
+def main(
+    expression: str,
+    genotype,
+    geneloc,
+    snploc,
+    covariates,
+    output_prefix: str,
+):
+    """
+    Creates a Hail Batch pipeline for calculating EQTLs
+    """
+    dataset = os.getenv('DATASET')
+    access_level = os.getenv('ACCESS_LEVEL')
+    backend = hb.ServiceBackend(
+        billing_project=dataset, bucket=f'cpg-{dataset}-{access_level}'
+    )
+    batch = hb.Batch(name='eQTL', backend=backend, default_python_image=DRIVER_IMAGE)
+
+    # load files into a python job to avoid memory issues during a submission
+    load_job = batch.new_python_job('load-data')
+    expression_df = load_job.call(pd.read_csv, expression, sep='\t')
```

Comment:
I'm confused a little bit about reading in files via click. I just ran this script and am getting the [error](https://batch.hail.populationgenomics.org.au/batches/7071/jobs/1):
`TypeError: get_number_of_scatters() missing 2 required positional arguments: 'expression_df' and 'geneloc_df'`
How do I actually reference the variables defined here (lines 195-199) correctly, such as on line 19 in the `def get_number_of_scatters()` function?

Reply:
It's kind of awkward, you correctly load the dataframes within the hail batch job, but you have to load it within the current script so that get_number_of_scatters has the literal value. You can see in the other script, we defined:

```py
print(f'Loading residuals: {residuals}')
residual_df_literal = pd.read_csv(residuals, sep='\t')
print(f'Loading significant_snps: {significant_snps}')
significant_snps_df_literal = pd.read_csv(
    significant_snps, sep=' ', skipinitialspace=True
)

# then you pass those two literal values as parameters to the script.
get_number_of_scatters(
    residual_df_literal, significant_snps_df_literal
)
```

Reply:
awesome, thanks that did it. The script now works!